client/lxd_instances.go

```diff
@@ -1249,10 +1253,16 @@ func (r *ProtocolLXD) ExecInstance(instanceName string, exec api.InstanceExecPos
 					return nil, err
 				}
 
+				go func() {
+					_, _, _ = conn.ReadMessage() // Consume pings from server.
```

Comment:
and over here pings are only consumed for non-interactive sessions ?!

expanding context, I cannot immediately understand if "interactive" branch of code just above needs to consume pings or not.

Reply:
Only the control channel (for both interactive and non-interactive) and the stdin channel for non-interactive sessions requires a conn.ReadMessage() call (which internally then loops reading websocket frames and consumes the ping messages). 

The other channels are already consuming ping messages because their direction is to read from the websocket and write to the stdout/stderr or the PTY.

I have observed this by setting up a custom ping handler on the client that logs when a ping was received and I was initially only seeing it for the stdout & stderr (both interactive and non-interactive), and not for the control and stdin for non-interactive sessions.